portal/src/graphql/portal/AuthenticationAuthenticatorSettings.tsx

```diff
@@ -20,7 +20,7 @@ import {
   SecondaryAuthenticatorType,
   PortalAPIApp,
 } from "../../types";
-import { isArrayEqualInOrder } from "../../util/misc";
+import { isArrayEqualInOrder, clearEmptyObject } from "../../util/misc";
```

Comment:
Can you explain why do we need to do this? Is it the rawConfig returned the server containing empty object and undefined?

Reply:
When constructing the new config, empty object may be generated.

1. New object may be added for optional field. However the added object is not used (eg. the whole section disabled, config not updated)
2. properties deleted, makes the parent object empty

So to avoid having to much branching (if... if ... create object), I created all the object needed, mutate the object using the screen state, the simplify the final config by removing undefined and empty object